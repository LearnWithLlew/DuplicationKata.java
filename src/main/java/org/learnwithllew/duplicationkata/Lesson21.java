package org.learnwithllew.duplicationkata;

public class Lesson21 extends Song
{
  public void singSong(int style, String... names)
  {
    switch (style)
    {
      case 1 :
        for (String name : names)
        {
          if (name.startsWith("L", 0))
          {
            sing("Hip Hip Horray! For " + name);
          }
          else
          {
            sing("Hello " + name + ", it's nice to meet you.");
          }
        }
        break;
      case 2 :
        for (String name : names)
        {
          if (name.startsWith("am", 1))
          {
            sing("Say yeah! Say yo! Say " + name);
          }
          else
          {
            sing("Hello " + name + ", it's nice to meet you.");
          }
        }
        break;
      case 3 :
        for (String name : names)
        {
          sing("Hello " + name + ", it's nice to meet you.");
        }
        break;
    }
  }
}
