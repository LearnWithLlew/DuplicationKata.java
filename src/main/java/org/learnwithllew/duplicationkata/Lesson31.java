package org.learnwithllew.duplicationkata;

public class Lesson31 extends Song
{
  public void singSong(int style, String... names)
  {
    switch (style)
    {
      case 1 :
        for (String name : names)
        {
          if (name.startsWith("L"))
          {
            sing("Hip Hip Horray! For " + name);
          }
          else
          {
            sing("Hello " + name + ", it's nice to meet you.");
          }
        }
        break;
      case 2 :
        for (String name : names)
        {
          if (name.contains("a"))
          {
            sing(name.toUpperCase() + "! Yay " + name + "!");
          }
          else
          {
            sing("Hello " + name + ", it's nice to meet you.");
          }
        }
        break;
      case 3 :
        for (String name : names)
        {
          sing("Hello " + name + ", it's nice to meet you.");
        }
        break;
    }
  }
}
