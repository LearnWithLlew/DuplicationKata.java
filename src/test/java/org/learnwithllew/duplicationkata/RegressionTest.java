package org.learnwithllew.duplicationkata;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.learnwithllew.duplicationkata.Lesson1Straight;
import org.learnwithllew.duplicationkata.Lesson21;
import org.learnwithllew.duplicationkata.Lesson2Variable;
import org.learnwithllew.duplicationkata.Lesson31;
import org.learnwithllew.duplicationkata.Lesson3HigherOrderFunctions;

public class RegressionTest
{
  @Test
  public void testCat()
  {
    Lesson1Straight song = new Lesson1Straight();
    song.singCatSong();
    Approvals.verify(song.song);
  }
  @Test
  public void testBeer()
  {
    Lesson2Variable song = new Lesson2Variable();
    song.singBottlesOfBeer();
    Approvals.verify(song.song);
  }
  @Test
  public void testNames()
  {
    Lesson21 song = new Lesson21();
    String[] names = {"Llewellyn", "Samantha", "Tomas", "Emilia"};
    song.singSong(1, names);
    song.singSong(2, names);
    song.singSong(3, names);
    Approvals.verify(song.song);
  }
  @Test
  public void testNumbers()
  {
    Lesson3HigherOrderFunctions song = new Lesson3HigherOrderFunctions();
    song.singCheers();
    Approvals.verify(song.song);
  }
  @Test
  public void testNames3()
  {
    Lesson31 song = new Lesson31();
    String[] names = {"Llewellyn", "Samantha", "Tomas", "Emilia"};
    song.singSong(1, names);
    song.singSong(2, names);
    song.singSong(3, names);
    Approvals.verify(song.song);
  }
}
