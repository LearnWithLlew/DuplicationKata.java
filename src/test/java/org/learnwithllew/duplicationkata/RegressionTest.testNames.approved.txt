Hip Hip Hurray! For Llewellyn
Hello Samatha, it's nice to meet you.
Hello Tomas, it's nice to meet you.
Hello Emilia, it's nice to meet you.
Hello Llewellyn, it's nice to meet you.
Say yeah! Say yo! Say Samatha
Hello Tomas, it's nice to meet you.
Hello Emilia, it's nice to meet you.
Hello Llewellyn, it's nice to meet you.
Hello Samatha, it's nice to meet you.
Hello Tomas, it's nice to meet you.
Hello Emilia, it's nice to meet you.
